enterprise_subsidy/apps/transaction/tests/test_management.py

```diff
@@ -0,0 +1,392 @@
+                    }],
+                    'entitlements': self.course_entitlements,
+                    'modified': '2022-05-26T15:46:24.355321Z',
+                    'additional_metadata': None,
+                    'enrollment_count': 0,
+                    'recent_enrollment_count': 0,
+                    'course_run_keys': [self.courserun_key],
+                    'content_last_modified': '2023-03-06T20:56:46Z',
+                    'enrollment_url': 'https://foobar.com',
+                    'active': False
+                },
+                200,
+            ),
+        ]
+        call_command('write_reversals_from_enterprise_unenrollments')
+        # Assert that we only make two calls with the oauth client, one to the enterprise service to fetch
+        # unenrollments and only one to the catalog service to fetch course metadata
+        assert mock_oauth_client.return_value.get.call_count == 2
+
+    @mock.patch('enterprise_subsidy.apps.api_client.base_oauth.OAuthAPIClient', return_value=mock.MagicMock())
+    def test_write_reversals_from_enterprise_unenrollment_transaction_does_not_exist(self, mock_oauth_client):
+        """
+        Test that the write_reversals_from_enterprise_unenrollments management command does not create a reversal if
+        the transaction does not exist.
+        """
+        mock_oauth_client.return_value.get.return_value = MockResponse(
+            [{
+                'enterprise_course_enrollment': {
+                    'enterprise_customer_user': 10,
+                    'course_id': self.courserun_key,
+                    'created': '2023-05-25T19:27:29Z',
+                    'unenrolled_at': '2023-06-1T19:27:29Z',
+                },
+                'transaction_id': uuid.uuid4(),
+                'uuid': self.fulfillment_identifier,
+            }],
+            200
+        )
+        assert Reversal.objects.count() == 0
+        call_command('write_reversals_from_enterprise_unenrollments')
+        assert Reversal.objects.count() == 0
+
+    @mock.patch('enterprise_subsidy.apps.api_client.base_oauth.OAuthAPIClient', return_value=mock.MagicMock())
+    @ddt.data(
+        ('2023-05-25T19:27:29Z', '2023-06-1T19:27:29Z'),
+        ('2023-06-1T19:27:29Z', '2023-05-25T19:27:29Z'),
+    )
+    @ddt.unpack
+    def test_write_reversals_from_enterprise_unenrollment_refund_period_ended(
+        self,
+        course_start_date,
+        enrollment_created_at,
+        mock_oauth_client,
+    ):
+        """
+        Test that for write_reversals_from_enterprise_unenrollments, if the greater date between the course start date
+        and the enrollment created at date is more than 14 days before the unenrollment date, no reversal is created.
+        """
+        # unenrolled_at is 14 days after the considered refund period start date so the reversal is not created
+        unenrolled_at = '2023-06-16T19:27:29Z'
+        mock_oauth_client.return_value.get.side_effect = [
```

Comment:
This sequencing of API calls feels pretty sensitive to changes.  Would it be possible instead to mock the individual subclasses of the base client, `EnterpriseCatalogApiClient` and `EnterpriseApiClient`?